### PR TITLE
GetCallingAssembly() skip past reflection. Fix #29108 - part 1

### PR DESCRIPTION
**Pull request by lambdageek**
https://bugzilla.xamarin.com/show_bug.cgi?id=29108#c6

### CODE REVIEW

**Comment by kumpera** [comment]
whitelist

**Comment by kumpera** [comment]
Hey,

It would be nice if we get a test case for this included in mcs/class/corlib.

File: mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	public static Assembly InvokeCall () {
+	    var o = typeof (Calls).GetMethod("Leaf").Invoke(null, null);
+	    sideEffect();
+	    return (Assembly)o;
+	}
+
+    }
+
+    [TestFixture]
+    public class GetCallingAssemblyTest
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs]: We don't do test fixture per issue. Please add this to existing tests

File: mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    {
+	// the name of this assembly
+	const String THIS_TEST_ASSEMBLY = "corlib_test_net_4_x";
+
+	[Test]
+	public void Simple () {
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs]: Please name new tests properly

File: mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	    var a = Calls.DirectCall ();
+	    Assert.IsNotNull (a);
+
+	    var n = a.GetName ();
+	    Assert.AreEqual (THIS_TEST_ASSEMBLY, n.Name);
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs]: why not to do Assert.AreEqual (GetType().Assembly, a) instead

Comment (lambdageek) [inline mcs/class/corlib/Test/System.Reflection/GetCallingAssemblyTest.cs]: Hm. Good idea.  Thanks!

**Comment by lambdageek** [comment]
Could we merge this? @marek-safar or @kumpera does this look good or should I rework it some more?

**Comment by marek-safar** [comment]
@lambdageek please merge once you rename test bug29108 to what it actually tests and resolve merge conflict

**Comment by lambdageek** [comment]
Ready. But I don't have write access to Mono.

**Pull request merged**